### PR TITLE
Add PNS application

### PR DESCRIPTION
**Pull request by jiangplus**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application and your interest in our grants program. The formatting of your application seems to be a little bit off, see “ that explain how a user can set social network identity of their PNS.” Could you fix this? Also, could you focus more on the technical details of what you want to develop and what you think is missing in the ecosystem? We usually don’t support the integration of a specific token (ksm for example), but rather want to support open source software that is beneficial for our ecosystem and might be used by other projects for something completely different.

**Comment by jiangplus** [comment]
> We usually don’t support the integration of a specific token (ksm for example), but rather want to support open source software that is beneficial for our ecosystem and might be used by other projects for something completely different.

Thank you for the question.

Since Polkadot is a multi-chain ecosystem, users may have different assets on different chains. As a DID infrastructure, PNS serves users across all parachains. If PNS only supports one token, e.g. GLMR, those without such tokens may find it difficult to use their DIDs. Thus, it is necessary to support multiple tokens to register PNS. This is not intended to promote a specific token, but to facilitate all Polkadot users at the ecosystem level.

**Comment by jiangplus** [comment]
Hi @Noc2 , we have added more tech details about the PNS application

**Comment by jiangplus** [comment]
Hi @Noc2 , we have updated our application to add more tech details and the reason why we think it is useful to the Polkadot ecosystem.

**Review by takahser — CHANGES_REQUESTED** [review]
@jiangplus thanks for adding more details.
Please note that deliverables 0a-0d of the [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) are usually mandatory for all milestones, and deliverable 0e at least for the last one. Could you add those that are missing?

File: applications/PNS.md [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT  |
+| 0b. | Documentation | We will provide documentation that explains how a user can register a domain with dot, ksm, aUSD, USDC.   |
+| 0c. | Tests | Core payment functions will be fully covered by unit tests to ensure functionality and robustness.  |
+| 1. | Multi-token support | Since Polkadot is a multi-chain ecosystem, users may have different assets on different chains. As a DID infrastructure, PNS serves users across all parachains. If PNS only supports one token, e.g. GLMR on Moonbeam, those without such tokens may find it difficult to use their DIDs. Thus, it is necessary to support multiple tokens to register PNS. This is not intended to promote a specific token, but to facilitate all Polkadot users at the ecosystem level. We will implement the functionality to accept dot, ksm to register PNS domains.  |
+| 2. | Interaction with Frontend and Payment ABI of price oracle | PNS registration and renewal fee is priced in USD. In order to display the price and receive the fee correctly, the PNS App should read the price of tokens from other oracles and allow users to pay various tokens to obtain their PNS.  |
```
Comment (takahser) [inline applications/PNS.md:300]: Did you already make plans on which oracle you're going to use and what the integration will look like?

Comment (jiangplus) [reply applications/PNS.md]: Hi @takahser,  we are using Chainlink data feeds as our price oracle to get the price of KSM/USD, DOT/USD and GLMR/USD.

**Comment by jiangplus** [comment]
@takahser 
Hi, we have updated the application with the required sections and our latest implementation plan.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the late response here. Could you still fix the formatting of your application? For example “ Social network identity integration” probably should be part of the table. Also, could you add more technical details to the deliveries? For example, how are you going to implement “Account ownership verification”? Do you implement this as part of the pallet or the front end or both?

**Comment by jiangplus** [comment]
@takahser @Noc2 
Hi, thanks for your reply.
I have added more tech details on social network identity verifications

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. I have a few follow-up questions/concerns:

- On the one hand the entire setup at the moment seems relatively centralised, how do you plan to change this? 
- On the other hand I would be much more interested in supporting the development of a pallet via the grants program or an ink! smart contract native to polkadot. 
- Also where are your current solidity smart contracts deployed (moonbeam)? In this case it might make much more sense to get a grant from moonbeam instead of us, since we don’t want to support a specific parachain and need to be objective.

**Comment by Noc2** [comment]
I’m closing this due to inactivity. Let me know if you want to re-open it.

**Comment by jiangplus** [comment]
Hi @Noc2 
We understand your concerns and are thinking about the roadmap of PNS and how it can be contributed to the Polkadot ecosystem.

Here are some replies to your questions:
1) At the moment, we are the provider of the credential auth service, but it doesn't have to be the case. The credential data and proofs can be accessed by anyone. And we are collaborating with [NextID](https://next.id/) to provide a general standard. Every service provider can provide their own verifiable credentials.

2,3 ) We have been working on PNS for more than one year and are building a well-polished product around the protocol. Beyond the functions of ENS, we are trying to build a flexible and multi-chain aware name system. It now deployed on Moonbeam that make it easier to connect with the EVM DApps. We are also building a name service pallet and a decentralized DNS module. We may apply the grant for that with a new application. The feature in this application is also very important to us and our users, and will also benefit the whole community of Polkadot as well.  It just takes a long time to design and build a fully decentralized system.

**Comment by Noc2** [comment]
Sorry for the late reply here. Definitely feel free to apply for a grant for the pallet or even an ink! smart contract implementation. Maybe you could also add certain XCM-based functions to this implementation.

**Pull request closed without merging**